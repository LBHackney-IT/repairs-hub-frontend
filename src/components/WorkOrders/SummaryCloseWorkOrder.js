import PropTypes from 'prop-types'
import { useForm } from 'react-hook-form'
import { PrimarySubmitButton } from '../Form'

const SummaryCloseWorkOrder = ({
  reference,
  onJobSubmit,
  notes,
  time,
  date,
  changeStep,
}) => {
  const { handleSubmit } = useForm({})

  return (
    <div>
      <h1 className="lbh-heading-h1">Update work order: {reference}</h1>
      <form role="form" onSubmit={handleSubmit(onJobSubmit)}>
        <h4 className="lbh-heading-h4">Summary of updates to work order</h4>
        <table className="govuk-table lbh-table">
          <tbody className="govuk-table__body">
            <tr className="govuk-table__row">
              <th scope="row" className="govuk-table__header">
                Completion time
              </th>
              <td className="govuk-table__cell">
                {date.split('-').join('/')} {time}
              </td>
              <td className="govuk-table__cell">
                <a className="lbh-link" onClick={changeStep} href="#">
                  Edit
                </a>
              </td>
            </tr>
            <tr className="govuk-table__row">
              <th scope="row" className="govuk-table__header">
                Notes
              </th>
              <td className="govuk-table__cell">{notes}</td>
              <td className="govuk-table__cell">
                <a className="lbh-link" onClick={changeStep} href="#">
                  Edit
                </a>
              </td>
            </tr>
          </tbody>
        </table>
        <PrimarySubmitButton label="Confirm and close" />
      </form>
    </div>
  )
}

SummaryCloseWorkOrder.propTypes = {
  reference: PropTypes.string.isRequired,
  onJobSubmit: PropTypes.func.isRequired,
  notes: PropTypes.string.isRequired,
  time: PropTypes.string,
  date: PropTypes.string,
  changeStep: PropTypes.func.isRequired,
}

export default SummaryCloseWorkOrder
