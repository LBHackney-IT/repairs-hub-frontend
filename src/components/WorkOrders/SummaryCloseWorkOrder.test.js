import { render } from '@testing-library/react'
import SummaryCloseWorkOrder from './SummaryCloseWorkOrder'

describe('SummaryCloseWorkOrder component', () => {
  const props = {
    reference: 10000012,
    notes: 'this is a note',
    time: '14:30',
    date: '2021-02-03T11:33:35.757339',
    reason: 'No Access',
    operatives: ['Operative A', 'Operative B'],
    onJobSubmit: jest.fn(),
    changeStep: jest.fn(),
  }

  it('should render properly', () => {
    const { asFragment } = render(
      <SummaryCloseWorkOrder
        reference={props.reference}
        notes={props.notes}
        time={props.time}
        date={props.date}
        reason={props.reason}
        onJobSubmit={props.onJobSubmit}
        changeStep={props.changeStep}
        operativeNames={props.operatives}
        isOvertime={true}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
