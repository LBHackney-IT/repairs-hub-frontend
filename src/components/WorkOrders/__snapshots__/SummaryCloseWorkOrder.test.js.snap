// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryCloseWorkOrder component should render properly 1`] = `
<DocumentFragment>
  <div>
    <h1
      class="lbh-heading-h1"
    >
      Close work order: 10000012
    </h1>
    <form
      role="form"
    >
      <h4
        class="lbh-heading-h4"
      >
        Summary of updates to work order
      </h4>
      <table
        class="govuk-table lbh-table"
      >
        <tbody
          class="govuk-table__body"
        >
          <tr
            class="govuk-table__row"
          >
            <th
              class="govuk-table__header"
              scope="row"
            >
              Completion time
            </th>
            <td
              class="govuk-table__cell"
            >
              2021/02/03T11:33:35.757339 14:30
            </td>
            <td
              class="govuk-table__cell"
            >
              <a
                class="lbh-link"
                href="#"
              >
                Edit
              </a>
            </td>
          </tr>
          <tr
            class="govuk-table__row"
          >
            <th
              class="govuk-table__header"
              scope="row"
            >
              Overtime
            </th>
            <td
              class="govuk-table__cell"
            >
              Yes
            </td>
            <td
              class="govuk-table__cell"
            >
              <a
                class="lbh-link"
                href="#"
              >
                Edit
              </a>
            </td>
          </tr>
          <tr
            class="govuk-table__row"
          >
            <th
              class="govuk-table__header"
              scope="row"
            >
              Operatives
            </th>
            <td
              class="govuk-table__cell"
            >
              Operative A, Operative B
            </td>
            <td
              class="govuk-table__cell"
            >
              <a
                class="lbh-link"
                href="#"
              >
                Edit
              </a>
            </td>
          </tr>
          <tr
            class="govuk-table__row"
          >
            <th
              class="govuk-table__header"
              scope="row"
            >
              Reason
            </th>
            <td
              class="govuk-table__cell"
            >
              No Access
            </td>
            <td
              class="govuk-table__cell"
            >
              <a
                class="lbh-link"
                href="#"
              >
                Edit
              </a>
            </td>
          </tr>
          <tr
            class="govuk-table__row"
          >
            <th
              class="govuk-table__header"
              scope="row"
            >
              Notes
            </th>
            <td
              class="govuk-table__cell"
            >
              this is a note
            </td>
            <td
              class="govuk-table__cell"
            >
              <a
                class="lbh-link"
                href="#"
              >
                Edit
              </a>
            </td>
          </tr>
        </tbody>
      </table>
      <div
        class="govuk-form-group lbh-form-group"
      >
        <button
          class="govuk-button lbh-button"
          data-module="govuk-button"
          type="submit"
        >
          Confirm and close
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
